release/models/cfm/openconfig-cfm.yang

```diff
@@ -0,0 +1,1286 @@
+      }
+      units "minutes";
+      description
+        "This object specifies the Measurement Interval for FLR statistics, in
+        minutes.
+        A Measurement Interval of 15 minutes needs to be supported,
+        other intervals may be supported.";
+      reference
+       "MEF-SOAM-PM-MIB.mefSoamLmCfgMeasurementInterval";
+    }
+
+    leaf repetition-period {
+      type uint32;
+      default 0;
+      description
+        "This object specifies a configurable repetition time between
+        Measurement Intervals in a Measurement session in hundredths of a
+        second.
+
+        If the value is less than or equal to one Measurement Interval
+        there is no time gap between the end of one Measurement
+        Interval and the start of a new Measurement Interval. This is
+        the normal usage case.
+
+        If the value is greater than one Measurement Interval there is
+        time gap between the end of one Measurement Interval and the
+        start of the next Measurement Interval.";
+      reference
+        "[MEF SOAM PM IA] R18, D3, R19, R20";
+    }
+
+    leaf intervals-archived {
+      type uint16;
+      description
+        "This object specifies the number of completed measurement intervals to
+        store in the history statistic table.
+
+        At least 32 completed measurement intervals are to be
+        stored. 96 measurement intervals are recommended to be stored.";
+      reference
+        "MEF-SOAM-PM-MIB.mefSoamLmCfgNumIntervalsStored";
+    }
+
+    leaf packets-per-meaurement-period {
+      type uint16;
+      description "Number of probe packets to send within a measurement interval.";
+    }
+
+    leaf burst-interval {
+      type uint32;
+      units "milliseconds";
+      description "Interval between bursts within a measurement interval.";
+    }
+
+    leaf packet-per-burst {
+      type uint32;
+      units "milliseconds";
+      description 
+      "if burst enabled. the leaf defines the number of packet per burst of probes.
+      packets-per-measurement-period / burst-interval";
+    }
+  }
+
+  grouping rdi-config {
+    description "Remote Defect Indication (RDI) configuration.";
+
+    leaf transmit-on-defect {
+      type boolean;
+      description "Transmit RDI when a defect is detected.";
+    }
+  }
+
+  grouping link-loss-forwarding-config {
+    description "Link loss forwarding configuration.";
+
+    leaf enable {
+      type boolean;
+      description 
+        "Enable propagation of the remote
+        attachment-circuit link state to the
+        local attachment-circuit link state";
+    }
+
+    leaf action {
+      type enumeration {
+        enum "SHUTDOWN" { description "Shutdown action on link loss."; }
+        enum "ALARM" { description "Raise an alarm on link loss."; }
+        enum "DISABLE_ROUTING" { description "Mark interface non-routable."; }
+      }
+      description "Action to take on link loss.";
+    }
+  }
+
+  uses cfm-top;
```

Comment:
Should there be some hierarchical structure to group OAM vs. CFM directly at the root?

Reply:
I support this to hopefully get a Juniper LFM augmentation.

Reply:
Yes, how about adding a top level container for `/oam` and nesting cfm under that as:  `/oam/cfm`

Reply:
So a specific proposal is to create a openconfig-oam.yang file with the top level /oam container.  Then another file for openconfig-oam-cfm.yang which augments the /oam path.